src/hocon.erl

```diff
@@ -171,6 +172,28 @@ do_abspath(Var, ['$root']) ->
 do_abspath(Var, [Path|More]) ->
     do_abspath(iolist_to_binary([atom_to_binary(Path, utf8), <<".">>, Var]), More).
 
+-spec expand(list()) -> {ok, list()}.
+expand(KVList) ->
+    {ok, do_expand(KVList, [])}.
+
+do_expand([], Acc) ->
+    lists:reverse(Acc);
+do_expand([{Key, {concat, C}}|More], Acc) ->
+    do_expand(More, [create_nested(Key, {concat, do_expand(C, [])})|Acc]);
+do_expand([{Key, Value}|More], Acc) ->
+    do_expand(More, [create_nested(Key, Value)|Acc]);
+do_expand([{Object}|More], Acc) when is_list(Object) ->
+    do_expand(More, [{do_expand(Object, [])}|Acc]);
+do_expand([Other|More], Acc) ->
+    do_expand(More, [Other|Acc]).
+
+create_nested(Key, Value) when is_atom(Key) ->
+    {concat, [{[Res]}]} = create_nested(paths(Key), Value),
+    Res;
+create_nested([], Value) ->
+    Value;
+create_nested([Path|More], Value) ->
+    {concat, [{[{Path, create_nested(More, Value)}]}]}.
```

Comment:
I am not quite following why it has to be a `concat`.

Reply:
As all objects are wrapped by `concat` in parser, we have to mock that to make it readable in the following pipeline.